transcript/src/main/java/com/transcriptanalyzer/transcript/Tree.java

```diff
@@ -71,17 +78,95 @@ public Node hasChild(String intent){
             }
             return null;
         }
+
+        // return a hashmap mapping the 3 leaf intents with most occurrences to their occurrences in descending order
+        public LinkedHashMap<String, Integer> getBestLeafData() {
+            // create a list for the best leaf data of all children
+            LinkedHashMap<String, Integer> leafData = new LinkedHashMap<>();
+            // create a placeholder list for the top three leaf intents
+            LinkedHashMap<String, Integer> bestLeafData = new LinkedHashMap<>();
```

Comment:
We might want to consider dividing up the process of collecting the best leaf data of all children and collecting the top three intents - these two variables have very similar names and method calls so it makes it difficult at times to track which one we're modifying at each step. Helper methods (i.e., one the returns the best leaf data for all children) would be useful here.